backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/ArtifactStoreService.java

```diff
@@ -114,4 +116,22 @@ default String getUrl(
 
     return uriComponentsBuilder.toUriString();
   }
+
+  default void validateArtifactSizeForTrial(App app, int artifactSize, Long partNumber)
+      throws ModelDBException {
+    if (app.getTrialEnabled()) {
+      if (partNumber != null && partNumber != 0) {
```

Comment:
we should return multipart upload false in trial mode

Reply:
this probably should be 1 https://github.com/VertaAI/modeldb/blob/master/client/verta/verta/_tracking/experimentrun.py#L272